Delete old projects and remove failing samples from the solution.

ahsonkhan opened the pull request:
Part of corefxlab repo cleanup

cc @dotnet/corefxlab-contrib, @davidfowl, @pakrym, @JeremyKuhne, @AtsushiKan, @adamsitnik

ahsonkhan commented on samples/LibuvWithNonAllocatingFormatters/LibuvWithNonAllocatingFormatters.csproj:
```diff
@@ -6,7 +6,6 @@
     <OutputType>Exe</OutputType>
```
Note that this sample was removed from the solution as it no longer builds (we removed System.Net.Libuv). What do we plan to do with this?

ahsonkhan commented on samples/System.IO.Pipelines.Samples/CompressionSample.cs:
```diff
@@ -4,8 +4,6 @@
 
```
Note that this sample was removed from the solution as it no longer builds (we removed all the System.IO.Pipelines.* projects). What do we plan to do with this?

cc @davidfowl, @pakrym

KrzysztofCwalina commented on samples/QotdService/QotdService.csproj:
```diff
@@ -5,7 +5,6 @@
     <OutputType>Exe</OutputType>
```
Hmmm, we need to port it to use Sockets. Please file an issue and assign it to me. I will do it.

ahsonkhan replied:
https://github.com/dotnet/corefxlab/issues/2250

KrzysztofCwalina commented on src/System.Numerics.Dec64/System.Numerics.Dec64.csproj:
```diff
@@ -1,10 +0,0 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Are we removing Dec64 for good or just moving it to some temporary location? If we are removing it, I think we need to contact the owner first.

ahsonkhan replied:
There is no steward within Microsoft for this.

@benaadams initiated it but it has been inactive since then:
https://github.com/dotnet/corefxlab/pull/764
https://github.com/dotnet/corefx/issues/10951

KrzysztofCwalina commented:
Most of the code removed in this PR was added by @davidfowl. I would like @davidfowl to sign off on the removals.

ahsonkhan commented on src/System.Numerics.Matrices/System.Numerics.Matrices.csproj:
```diff
@@ -1,9 +0,0 @@
-<Project Sdk="Microsoft.NET.Sdk">
```
Similar to System.Numerics.Dec64, we are deleting System.Numerics.Matrices.

There is no steward/owner and the project has become inactive.

cc @KrzysztofCwalina, @whoisj 

It was added here: https://github.com/dotnet/corefxlab/pull/118

davidfowl commented:
RIP 👍

davidfowl commented:
Did you delete the pipelines samples?

ahsonkhan commented:
> Did you delete the pipelines samples?

No, I just removed it from the solution. It no longer builds because we removed the other System.IO.Pipelines.* projects.

davidfowl commented:
> No, I just removed it from the solution. It no longer builds because we removed the other System.IO.Pipelines.* projects. Should I remove it?

Yes, nuke it.
